### ISSUE TITLE
ofVideoPlayer isPlaying(), different behavior on ofVideoPlayer with or without Gstreamer

### ISSUE THREAD

**Issue by dimitre** [openframeworks/openFrameworks#6527]
Using Gstreamer isPlaying() always returns true even when paused and using plain ofVideoPlayer it returns false when paused. It would be great to have a standard behavior.
see:
https://github.com/arturoc/ofxGStreamer/issues/27

**Comment by roymacdonald** [comment]
Hi, I came across this issue as well when fixing [this](https://github.com/openframeworks/openFrameworks/pull/6517), although as I was digging under ofxGStreamer's hood, I figured out what @arturoc  [explained](https://github.com/arturoc/ofxGStreamer/issues/27), regardless of this it should be fixed so it makes sense.

**Comment by ofTheo** [comment]
I think @arturoc is correct. 
would be good to standardize across platforms. 

I can take a look at macOS and Windows.  
Can we assume streamer is fine as is?

**Comment by roymacdonald** [comment]
I am not sure what it correct or fine in this case. 
Having a function that is called `isPlaying()`which returns true even when it is paused feels a bit counter-intuitive. 
is there any other big difference between stopping and pausing a video player other than that the former moves the playhead back to the begining? 
calling `stop()`, as fas as I understand does not deallocate memory, thus, from the users perspective it is not really different  from calling `videoPlayer.setPaused(true);videoPlayer.setPosition(0);`. 
I think that `isPlaying()` should simply be the negation of `isPaused()` 
I did just a quick check and there is no consistent behavior in this regard. 
I agree upon standarizing across platforms but, which would be this standard?

**Comment by dimitre** [comment]
yes, to me it is more intuitive that isPlaying returns false when paused.

**Comment by ofTheo** [comment]
Hmm. 
I always understood isPlaying() to be related to play() stop() and isPaused() for toggling pause states. 

I think it makes sense that isPlaying is true unless stop has been called.

**Comment by dimitre** [comment]
|  | isPlaying()  | isPaused()  |
|---|---|---|
| stopped  | false  | ?  |
| paused  | ?  | true  |
| playing  | true  | false  |

**Comment by ofTheo** [comment]
okay this seems to be the last issue on #6483 

Here are the different implementations currently. 
Bolded for values that aren't the same. 

**GStreamer:**

|  | isPlaying()  | isPaused()  |
|---|---|---|
| stopped  | false  | **true**  |
| paused  | **true**  | true  |
| playing  | true  | false | 


**macOS:**  [isPaused is the inverse of isPlaying](https://github.com/openframeworks/openFrameworks/blob/patch-release/libs/openFrameworks/video/ofAVFoundationPlayer.mm#L577 )

|  | isPlaying()  | isPaused()  |
|---|---|---|
| stopped  | false  | **true**  |
| paused  | **false**  | true  |
| playing  | true  | false | 


**Windows:**

|  | isPlaying()  | isPaused()  |
|---|---|---|
| stopped  | false  | **false**  |
| paused  | **false**  | true  |
| playing  | true  | false | 


Edit: for reference Processing does false/false for stopped: 
https://github.com/processing/processing-video/blob/master/src/processing/video/Movie.java#L346-L353

And false / true for pause: 
https://github.com/processing/processing-video/blob/master/src/processing/video/Movie.java#L346-L353 

So basically the Windows model. 

I am nervous though that any changes we might do might break projects and especially addons. 
I would maybe lean towards Processing / OF Windows approach - but def open to thoughts.

**Comment by dimitre** [comment]
great @ofTheo . 
maybe considering next release 0.11.1 or something, so it doesn't break projects based on release?
I think the windows model makes sense if you imagine a cassette tape player. paused is different than stopped. 
for me it makes no sense that a video paused returns isPlaying() as true.

**Comment by ofTheo** [comment]
Thanks! That's a good point @dimitre 
0.11.1 is def meant to be more of a patch release and breaking changes typically shouldn't be in those. 

I think targeting this for 0.12.0 makes more sense.

**Comment by arturoc** [comment]
One thing that we could do to avoid breaking old projects in hard to detect ways could be to deprecate the isPlaying isPaused API and change it with something else uniform across platforms and easier to understand. Perhaps a getState() that returns an enum with values Stopped, Paused, Playing?  The current isPlaying /isPaused would be deprecated but untouched.

**Comment by arturoc** [comment]
But yeah agree with dimitre that any api changes should go in 0.12 not a patch release

**Comment by dimitre** [comment]
I would vote for changing behavior even with potential to break previous projects. 
Or deprecating it but reflecting on documentation, ofBook etc. 
Because it is frustrating for beginners to have different behavior by platform, it is hard to realize it is not your fault.

**Comment by dimitre** [comment]
> I would maybe lean towards Processing / OF Windows approach - but def open to thoughts.

+1 for this approach. it seems the most correct to me. 
I think it should be corrected in the regular functions even with potential to break previous projects. 
it can be painful at first but it is best on the long run.

**Comment by ofTheo** [comment]
closed by #6987 #6986
